cosmwasm/msg_parser.go

```diff
@@ -0,0 +1,63 @@
+package cosmwasm
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/CosmWasm/wasmd/x/wasm"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	poststypes "github.com/desmos-labs/desmos/v3/x/posts/types"
+	profilestypes "github.com/desmos-labs/desmos/v3/x/profiles/types"
+	reactionstypes "github.com/desmos-labs/desmos/v3/x/reactions/types"
+	relationshipstypes "github.com/desmos-labs/desmos/v3/x/relationships/types"
+	reportstypes "github.com/desmos-labs/desmos/v3/x/reports/types"
+	subspacestypes "github.com/desmos-labs/desmos/v3/x/subspaces/types"
+)
+
+const (
+	WasmMsgParserRouteProfiles      = profilestypes.ModuleName
+	WasmMsgParserRouteSubspaces     = subspacestypes.ModuleName
+	WasmMsgParserRouteRelationships = relationshipstypes.ModuleName
+	WasmMsgParserRoutePosts         = poststypes.ModuleName
+	WasmMsgParserRouteReports       = reportstypes.ModuleName
+	WasmMsgParserRouteReactions     = reactionstypes.ModuleName
+)
+
+type MsgParserInterface interface {
+	Parse(contractAddr sdk.AccAddress, msg wasmvmtypes.CosmosMsg) ([]sdk.Msg, error)
+	ParseCustomMsgs(contractAddr sdk.AccAddress, data json.RawMessage) ([]sdk.Msg, error)
+}
+
+type ParserRouter struct {
+	Parsers map[string]MsgParserInterface
+}
+
+func NewParserRouter() ParserRouter {
+	return ParserRouter{
+		Parsers: make(map[string]MsgParserInterface),
+	}
+}
+
+type CustomMsg struct {
+	Route   string          `json:"route"`
+	MsgData json.RawMessage `json:"msg_data"`
+}
+
+func (router ParserRouter) ParseCustom(contractAddr sdk.AccAddress, data json.RawMessage) ([]sdk.Msg, error) {
+	var customMsg CustomMsg
+	err := json.Unmarshal(data, &customMsg)
+
+	log.Println("[!] CosmWasm contract msg routed to module: ", customMsg.Route)
```

Comment:
The message is for debug, it would be better to be `log.Debug`, instead.

Reply:
The `Debug` function doesn't exist for the `log` package.

Reply:
Okay, can we remove it instead?

Reply:
Yes I think so, it was for test purposes anyway